otelzap/otelzap.go

```diff
@@ -139,26 +139,20 @@ func (l *Logger) FatalContext(ctx context.Context, msg string, fields ...zapcore
 func (l *Logger) logFields(
 	ctx context.Context, lvl zapcore.Level, msg string, fields []zapcore.Field,
 ) []zapcore.Field {
+	allFields := append(append([]zapcore.Field{}, l.extraFields...), fields...)
```

Comment:
For better performance, let's rewrite this so we make a copy only when there are `l.extraFields`.

Also, no need to introduce a new variable `allFields`. We can continue to use `fields` name.

Reply:
Updated